src/lib/__tests__/downloadReachability-test.js

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable no-unused-expressions */
+import 'whatwg-fetch';
+import fetchMock from 'fetch-mock';
+import * as dr from '../downloadReachability';
+
+const sandbox = sinon.sandbox.create();
+
+const DEFAULT_DOWNLOAD_HOST_PREFIX = 'https://dl.';
+const DOWNLOAD_NOTIFICATION_SHOWN_KEY = 'download_host_notification_shown';
+const DOWNLOAD_HOST_FALLBACK_KEY = 'download_host_fallback';
+
+describe('lib/downloadReachability', () => {
+    beforeEach(() => {
+        sessionStorage.clear();
+        localStorage.clear();
+
+    });
+    
+    afterEach(() => {
+        sessionStorage.clear();
+        localStorage.clear();
+        sandbox.verifyAndRestore();
+
+    });
+    
+    describe('isCustomDownloadHost()', () => {
+        it('should be true if the url does not start with the default host prefix but is a dl host', () => {
+            let url = 'https://dl3.boxcloud.com/foo';
+            let result = dr.isCustomDownloadHost(url)
+            expect(result).to.be.true;
+
+            url = 'https://dl.boxcloud.com/foo';
+            expect(dr.isCustomDownloadHost(url)).to.be.false;
+
+            url = 'https://www.google.com';
+            expect(dr.isCustomDownloadHost(url)).to.be.false;
+
+
+            url = 'https://kld3lk.boxcloud.com';
+            expect(dr.isCustomDownloadHost(url)).to.be.true;
+
+            url = 'https://dl3.user.inside-box.net';
+            expect(dr.isCustomDownloadHost(url)).to.be.true;
+
+
+            url = 'https://dl.user.inside-box.net';
+            expect(dr.isCustomDownloadHost(url)).to.be.false;
+        });
+    });
+
+    describe('replaceDownloadHostWithDefault()', () => {
+        it('should add the given host to the array of shown hosts', () => {
+            const blockedHost = 'https://dl3.boxcloud.com';
+
+            const result = dr.setDownloadHostNotificationShown(blockedHost);
+
+            const shownHostsArr = JSON.parse(localStorage.getItem(DOWNLOAD_NOTIFICATION_SHOWN_KEY)) || [];
+            expect(shownHostsArr).to.contain('https://dl3.boxcloud.com');
+        });
+    });
+        
+    describe('setDownloadHostFallback()', () => {
+        it('should set the download host fallback key to be true', () => {
+            expect(sessionStorage.getItem(DOWNLOAD_HOST_FALLBACK_KEY) === 'true').to.be.false;
```

Comment:
This looks a bit weird

expect(sessionStorage.getItem(DOWNLOAD_HOST_FALLBACK_KEY)).to.equal('true')